src/ucp/wireup/address.c

```diff
@@ -502,11 +503,28 @@ ucp_address_unpack_iface_attr(ucp_worker_t *worker,
         return UCS_OK;
     }
 
-    packed                = ptr;
-    iface_attr->priority  = packed->prio_cap_flags & UCS_MASK(8);
-    iface_attr->overhead  = packed->overhead;
-    iface_attr->bandwidth = packed->bandwidth;
-    iface_attr->lat_ovh   = packed->lat_ovh;
+    packed               = ptr;
+    iface_attr->priority = packed->prio_cap_flags & UCS_MASK(8);
+    iface_attr->overhead = packed->overhead;
+    iface_attr->lat_ovh  = packed->lat_ovh;
+
+    if (packed->bandwidth < 0.0) {
+        /* The received value of the bandwidth is "dedicated - shared" which
+         * doesn't consider ppn and could be sent only by the peer which uses
+         * some old version of UCX. Calculate the bandwidth value considering
+         * our ppn to be compatible with the peer */
+        bandwidth.shared      = fabs(packed->bandwidth);
+        bandwidth.dedicated   = 0.0;
+        iface_attr->bandwidth = ucp_tl_iface_bandwidth(worker->context,
+                                                       &bandwidth);
+    } else {
+        /* The received value of the bandwidth is either the locally calculated
+         * bandwidth which considers ppn or the dedicated bandwidth sent by the
+         * peer which uses some old version of UCX */
```

Comment:
the received value is either a dedicated bandwidth value (when the peer is using an older version), or the total bandwidth considering remote ppn value (when the peer is using a newer version)

Reply:
done